x-pack/plugins/stack_connectors/common/crowdstrike/schema.ts

```diff
@@ -0,0 +1,199 @@
+                applied_date: schema.string(),
+                uninstall_protection: schema.string(),
+              }),
+              global_config: schema.object({
+                policy_type: schema.string(),
+                policy_id: schema.string(),
+                applied: schema.boolean(),
+                settings_hash: schema.string(),
+                assigned_date: schema.string(),
+                applied_date: schema.string(),
+              }),
+              remote_response: schema.object({
+                policy_type: schema.string(),
+                policy_id: schema.string(),
+                applied: schema.boolean(),
+                settings_hash: schema.string(),
+                assigned_date: schema.string(),
+                applied_date: schema.string(),
+              }),
+            }),
+            groups: schema.arrayOf(schema.any()),
+            group_hash: schema.string(),
+            product_type_desc: schema.string(),
+            provision_status: schema.string(),
+            serial_number: schema.string(),
+            status: schema.string(),
+            system_manufacturer: schema.string(),
+            system_product_name: schema.string(),
+            tags: schema.arrayOf(schema.any()),
+            modified_timestamp: schema.any(),
+            meta: schema.object({
+              version: schema.string(),
+              version_string: schema.string(),
+            }),
+            zone_group: schema.string(),
+            kernel_version: schema.string(),
+            chassis_type: schema.string(),
+            chassis_type_desc: schema.string(),
+            connection_ip: schema.string(),
+            default_gateway_ip: schema.string(),
+            connection_mac_address: schema.string(),
+            linux_sensor_mode: schema.string(),
+            deployment_type: schema.string(),
+          },
+          { unknowns: 'allow' }
+        )
+      )
+    ),
+    errors: schema.nullable(schema.arrayOf(schema.any())),
+    meta: schema.object({
+      query_time: schema.number(),
+      powered_by: schema.string(),
+      trace_id: schema.string(),
+    }),
+  },
+  { unknowns: 'allow' }
+);
+export const CrowdstrikeHostActionsResponseSchema = schema.object(
+  {
+    resources: schema.arrayOf(
+      schema.object({
+        id: schema.string(),
+        path: schema.string(),
+      })
+    ),
+    meta: schema.object({
+      query_time: schema.number(),
+      powered_by: schema.string(),
+      trace_id: schema.string(),
+    }),
+    errors: schema.nullable(schema.arrayOf(schema.any())),
+  },
+  { unknowns: 'allow' }
+);
+
+export const CrowdstrikeHostActionsParamsSchema = schema.object({
+  command: schema.oneOf([schema.literal('contain'), schema.literal('lift_containment')]),
+  actionParameters: schema.maybe(schema.object({}, { unknowns: 'allow' })),
+  ids: schema.arrayOf(schema.string()),
+  alertIds: schema.maybe(schema.arrayOf(schema.string())),
```

Comment:
I have trouble understanding. Are you saying that the subactions we invoke are stored in saved objects somewhere? 
I think we do not need that for our needs (@paul-tavares please correct me if I am wrong), so I will adjust the code to your preference. 
Should I then wrap all of them in schema.nullable() ?

Reply:
During normal rule execution, if a rule triggers an action, the params for that action are stored in an `action_task_params` saved object and then the action executor, whenever it runs, will read this saved object from persistence. 

The other way of executing actions is by directly calling `execute` via the `actionsClient` and this does not require this extra saved object layer. I'm guessing that the bidirectional connectors will always be executed this second way which doesn't require the saved object? If that's the case, the `schema.maybe` seems fine

Reply:
Correct. for now, we are not plugged into the whole "rule" processing. We're using the connector directly via the `ActionClient` and `UnsecuredActionClient`

@ymao1 - thanks for that explanation. I had been wondering they I kept seeing so much use of `schema.nullable()` instead of `schema.maybe()`.